src/lib/voiceSearchHelper/__tests__/index-test.ts

```diff
@@ -191,4 +191,19 @@ describe('VoiceSearchHelper', () => {
     voiceSearchHelper.dispose();
     expect(stop).toHaveBeenCalledTimes(1);
   });
+
+  it('stops and the status becomes `finished`', () => {
+    window.SpeechRecognition = createFakeSpeechRecognition();
+    const onQueryChange = (): void => {};
+    const onStateChange = (): void => {};
+    const voiceSearchHelper = createVoiceSearchHelper({
+      searchAsYouSpeak: true,
+      onQueryChange,
+      onStateChange,
+    });
+
+    voiceSearchHelper.toggleListening();
+    voiceSearchHelper.toggleListening();
```

Comment:
IIIRC the main problem was that `setState` was called multiple times with different values. From the `onEnd` listener and inside `stop` (via `resetState`). From what I've understood, now they are called with the same values. But the function is still called two times no?

Reply:
Yes, the `status` changed like this:
`initial` -> `waiting` -> (button click) -> `initial` (which shouldn't be) -> `finished`.

It's supposed to be like this:
`initial` -> `waiting` -> (button click) -> `finished`.

Now with these commits, after recognition started, if user clicks the button (which triggers `stop`),
the `stop()` will call `dispose()` and `resetState('finished');`.

`dispose()` will simply stop the recognition and remove all the event listeners. Now that the listeners are gone, `onEnd` will not be called.
and `resetState('finished')` will set the `status` to `'finished'`.

To sum up, in this version, after `(button click)`, `setState` is called only once.

Reply:
Yes, thanks!